x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -60,6 +65,18 @@ interface BaseProps {
    * The optional test subject associated with this DOM element.
    */
   'data-test-subj'?: string;
+
+  /**
+   * Indicates to the user whether the currently dragged item
+   * will be moved or copied
+   */
+  dragType?: 'copy' | 'move';
+
+  /**
+   * Indicates to the user whether the drop action will
+   * replace something that is existing or add a new one
+   */
+  dropType?: 'add' | 'replace';
```

Comment:
This new API looks good to me, it's the direction I was planning on taking it once we support "partial" operations!